tests/helpers/testers.py

```diff
@@ -159,13 +167,17 @@ def _functional_test(
         metric_functional: lightning metric functional that should be tested
         sk_metric: callable function that is used for comparison
         metric_args: dict with additional arguments used for class initialization
+        kwargs_update: Additional keyword arguments that will be passed with preds and
+            target when running update on the metric.
     """
     if not metric_args:
         metric_args = {}
+
     metric = partial(metric_functional, **metric_args)
 
     for i in range(NUM_BATCHES):
-        lightning_result = metric(preds[i], target[i])
+        extra_kwargs = {k: v[i] for k, v in kwargs_update.items()}
```

Comment:
same as above

Reply:
update in https://github.com/PyTorchLightning/metrics/pull/137/commits/86e4015cb0bf85bd10c7f860ada0c8a0c93124d5